x/auth/tx/service_test.go

```diff
@@ -145,8 +145,14 @@ func (s IntegrationTestSuite) TestSimulateTx_GRPC() {
 			} else {
 				s.Require().NoError(err)
 				// Check the result and gas used are correct.
-				s.Require().Equal(len(res.GetResult().GetEvents()), 6) // 1 coin recv 1 coin spent, 1 transfer, 3 messages.
-				s.Require().True(res.GetGasInfo().GetGasUsed() > 0)    // Gas used sometimes change, just check it's not empty.
+				//
+				// The 13 events are:
+				// - Sending Fee to the pool: coin_spent, coin_received, transfer and message.sender=<val1>
+				// - tx.* events: tx.fee, tx.acc_seq, tx.signature
+				// - Sending Amount to recipient: coin_spent, coin_received, transfer and message.sender=<val1>
+				// - Msg events: message.module=bank and message.action=/cosmos.bank.v1beta1.MsgSend
+				s.Require().Equal(len(res.GetResult().GetEvents()), 13) // 1 coin recv 1 coin spent, 1 transfer, 3 messages.
```

Comment:
So seems like the story goes:
- in v0.45, we had 6 events (probably a bug?)
- in v0.46 middleware refactor, we fixed this to return 13 events: https://github.com/cosmos/cosmos-sdk/pull/9920#discussion_r692184759
- we reverted to v0.45 code in #11979, so reverted to 6 events
- but now actually we're back to 13 events with this PR

Anyways, 13 events seems more correct to me, so 🤷